server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -90,6 +90,12 @@ public Symbol normalizeSymbol(Function symbol, TransactionContext txnCtx, NodeCo
         if (arg instanceof Input<?> input) {
             return Literal.of(input.value() == null);
         }
+        if (arg instanceof Reference ref
+            // allow system columns for further error processing, for example see testSelectWhereVersionIsNullPredicate().
+            && !ref.column().isSystemColumn()
+            && !ref.isNullable()) {
+            return Literal.of(false);
+        }
```

Comment:
This normalization is limited to columns only, need to expand it to handle expressions, ex) `<not_null_ref> + 1 is null`.
This also reminds me of https://github.com/crate/crate/blob/efeb10240fe14e11141b82019e2e9d381c4e3da5/server/src/main/java/io/crate/expression/predicate/NotPredicate.java#L184-L196 which also does something similar(but for `Collect` operation not `Get` as in the original bug report).
Once the proper normalization is in place, it should be able to be removed.